e2e/cp_test.go

```diff
@@ -4171,3 +4172,32 @@ func TestLocalFileOverridenWhenDownloadFailed(t *testing.T) {
 	expected := fs.Expected(t, fs.WithFile(filename, content))
 	assert.Assert(t, fs.Equal(workdir.Path(), expected))
 }
+
+// Test that counting writer does not corrupt objects during a download process
+func TestCountingWriter(t *testing.T) {
+	t.Parallel()
+
+	const (
+		filename = "log.txt"
+		size     = 3 * 1024 * 1024
+	)
+	content := make([]byte, size)
+	rand.Read(content)
+
+	s3client, s5cmd := setup(t)
+	bucket := s3BucketFromTestName(t)
+	createBucket(t, s3client, bucket)
+	putFile(t, s3client, bucket, filename, string(content))
+
+	workdir := fs.NewDir(t, t.Name())
+	defer workdir.Remove()
```

Comment:
explicit workdir should not be necessary. please take a look at `TestCopySingleS3ObjectToLocal` test.